### DISCUSSION TITLE
Avalonia Telemetry and GDPR

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#11183] [General]
From Avalonia 11 preview 7+ The team decided to activate telemetry, this could cause some violations of the European rules of the GDPR.

Some of the probable violations:
- it is not shown which data are collected (e.g. Avalonia version, email, etc..)
- who is responsible for the treatment
- how the data is processed and for what purpose
- how they are archived and where (regulation requires them to be archived in Europe for European users)
- consent to data processing is not required

**Comment by Gillibald** [comment]
https://github.com/AvaloniaUI/Avalonia/pull/11116#issue-1681751806

The information what gets collected is here

There is no personal information collected

**Reply by workgroupengineering** [reply]
The name of the project could be considered sensitive data because its purpose could be guessed from the name. As well possibly as the mail address or id of the device on which the build is performed.

**Reply by MikeCodesDotNET** [reply]
> The name of the project could be considered sensitive data because its purpose could be guessed from the name. As well possibly as the mail address or id of the device on which the build is performed.

We don't know the name of the project as it gets hashed before being sent. The reason we collect the hashed name is to understand the total number of projects being developed. We've zero interest in the actual name of projects.

**Comment by MikeCodesDotNET** [comment]
Hi, 

We’ve added anonymous telemetry to P7 with a [recent PR](https://github.com/AvaloniaUI/Avalonia/pull/11116). We did this to understand better how Avalonia is used and improve it for everyone. We’ve followed industry norms, looking at the likes of Uno for inspiration on how and what data is collected. You can read more about our Telemetry collection on [our wiki](https://github.com/AvaloniaUI/Avalonia/wiki/Telemetry). One thing that was essential for us in creating the telemetry system was not to include any identifiable information. 

With that said, If you wish to opt out of the telemetry, you can set the following environment variable to 1:
AVALONIA_TELEMETRY_OPTOUT

## To answer your questions: 

### it is not shown which data are collected (e.g. Avalonia version, email, etc..)
_‘Hashed’ means the data is anonymised using SHA256._

**We record the following information:** 
* Timestamp of the build
* Hashed project name. 
* Hashed git email address. 
* A unique but anonymous GUID for machine id.
* IDE Used (Visual Studio, VS4Mac, Rider or CLI) 
* CI Provider (Gitlab CI, Teamcity, etc, identified by environment variables)
* Project output type
* TFM 
* Avalonia version
* OS Version
* Architecture 

### who is responsible for the treatment
The telemetry is only accessible to the leadership of Avalonia UI and isn’t shared with any 3rd parties. 

The data is processed to display telemetry information in a dashboard. We look at the following: 
* Total number of unique projects. 
* Total number of users
* Total number of builds 
* IDE usage 
* CI Provider usage
* OS Versions

### how they are archived and where (regulation requires them to be archived in Europe for European users)
All data is stored in Azure, West Europe, using Azure Table Storage and is encrypted using 256-bit AES encryption. The plan is to wipe the data every 30 days after summarising the key metrics we’re interested in.

## To reiterate: We do not collect any identifiable information from Avalonia users.
